engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/UniverseSetupScreen.java

```diff
@@ -188,10 +243,118 @@ private void recursivelyAddModuleDependencies(Set<Name> modules, Name moduleName
+
+            EnvironmentSwitchHandler environmentSwitcher = new EnvironmentSwitchHandler();
+            context.put(EnvironmentSwitchHandler.class, environmentSwitcher);
+
+            environmentSwitcher.handleSwitchToPreviewEnvironment(context, environment);
+        }
+    }
+
+    private void initAssets() {
+        DefaultBlockFamilyFactoryRegistry familyFactoryRegistry = new DefaultBlockFamilyFactoryRegistry();
+        context.put(BlockFamilyFactoryRegistry.class, familyFactoryRegistry);
+
+        // cast lambdas explicitly to avoid inconsistent compiler behavior wrt. type inference
+        assetTypeManager.registerCoreAssetType(Prefab.class,
+                (AssetFactory<Prefab, PrefabData>) PojoPrefab::new, false, "prefabs");
+        assetTypeManager.registerCoreAssetType(BlockShape.class,
+                (AssetFactory<BlockShape, BlockShapeData>) BlockShapeImpl::new, "shapes");
+        assetTypeManager.registerCoreAssetType(BlockSounds.class,
+                (AssetFactory<BlockSounds, BlockSoundsData>) BlockSounds::new, "blockSounds");
+        assetTypeManager.registerCoreAssetType(BlockTile.class,
+                (AssetFactory<BlockTile, TileData>) BlockTile::new, "blockTiles");
+        assetTypeManager.registerCoreAssetType(BlockFamilyDefinition.class,
+                (AssetFactory<BlockFamilyDefinition, BlockFamilyDefinitionData>) BlockFamilyDefinition::new, "blocks");
+        assetTypeManager.registerCoreFormat(BlockFamilyDefinition.class,
+                new BlockFamilyDefinitionFormat(assetTypeManager.getAssetManager(), familyFactoryRegistry));
+        assetTypeManager.registerCoreAssetType(UISkin.class,
+                (AssetFactory<UISkin, UISkinData>) UISkin::new, "skins");
+        assetTypeManager.registerCoreAssetType(BehaviorTree.class,
+                (AssetFactory<BehaviorTree, BehaviorTreeData>) BehaviorTree::new, false, "behaviors");
+        assetTypeManager.registerCoreAssetType(UIElement.class,
+                (AssetFactory<UIElement, UIData>) UIElement::new, "ui");
+
+    }
+
+    /**
+     * Create a list of the names of the world, so that they can be displayed as simple String
+     * in the drop-down.
+     * @return A list of world names in String.
+     */
+    public List<String> worldNames() {
+        List<String> worldNamesList = Lists.newArrayList();
+        for (WorldSetupWrapper world : worlds) {
+            worldNamesList.add(world.getWorldName().toString());
+        }
+        return worldNamesList;
+    }
+
+    /**
+     * This method takes the name of the selected world as String and return the corresponding
+     * WorldSetupWrapper object.
+     * @return {@link WorldSetupWrapper} object.
+     */
+    public WorldSetupWrapper findWorldByName() {
+        for (WorldSetupWrapper world : worlds) {
+            if (world.getWorldName().toString().equals(selectedWorld)) {
+                return world;
+            }
+        }
+        return null;
+    }
+
+    /**
+     *
```

Comment:
Description :D

In the case of generic setter/getters it might make sense to skip javadoc

Reply:
Skipping java doc then.